conan/cli/commands/graph.py

```diff
@@ -298,3 +300,118 @@ def graph_explain(conan_api, parser,  subparser, *args):
+    cwd = os.getcwd()
+    path = conan_api.local.get_conanfile_path(args.path, cwd, py=None) if args.path else None
+
+    # Basic collaborators, remotes, lockfile, profiles
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile,
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates,
+                                                         is_build_require=args.build_require)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update,
+                                                         check_updates=args.check_updates)
+    print_graph_basic(deps_graph)
+    ConanOutput().title("Checking remotes")
+
+    # Data structure to store info per library
+    dict_nodes = {}
+    dependencies = deps_graph.nodes[1:]
+    for node in dependencies:
+        dict_nodes.setdefault(node.name, {"cache_refs": set(), "version_ranges": [],
+                                          "latest_remote": None})["cache_refs"].add(node.ref)
+
+    resolved_ranges = list(deps_graph.resolved_ranges.keys())
+    for version_range in resolved_ranges:
+        if version_range.name in dict_nodes:
+            dict_nodes[version_range.name]["version_ranges"].append(version_range)
+
+    # find in remotes
+    _find_in_remotes(conan_api, dict_nodes, remotes)
+
+    _remove_not_outdated_libs(dict_nodes)
+
+    return dict_nodes
+
+
+def _remove_not_outdated_libs(dict_nodes):
+    not_outdated = []
+    for node_name, node in dict_nodes.items():
+        if node['latest_remote'] is None or (len(node['cache_refs']) == 1
+                and list(node['cache_refs'])[0] >= node['latest_remote']['ref']):
+            not_outdated.append(node_name)
+    for node_name in not_outdated:
+        del dict_nodes[node_name]
+
+
+def _find_in_remotes(conan_api, dict_nodes, remotes):
+    for node_name in dict_nodes.keys():
```

Comment:
```suggestion
    for node_name, node_info in dict_nodes.items():
```
So you don't have to access later with ``dict_nodes[node_name]``, but use ``node_info`` directly